conformance_tests/test_tiovx/test_ext_supplementary_data.c

```diff
@@ -0,0 +1,1435 @@
+            vxReleaseImage(&image);
+            break;
+        }
+        case VX_TYPE_PYRAMID:
+            ref = (vx_reference)vxCreatePyramid(context, 2, VX_SCALE_PYRAMID_HALF, 16, 16, VX_DF_IMAGE_U8);
+            break;
+        case VX_TYPE_SCALAR:
+        {
+            vx_uint32 value = 0;
+            ref = (vx_reference)vxCreateScalar(context, VX_TYPE_UINT32, &value);
+            break;
+        }
+        case VX_TYPE_TENSOR:
+        {
+            vx_size dims[3] = {4, 4, 4};
+            ref = (vx_reference)vxCreateTensor(context, 3, dims, VX_TYPE_INT32, 0);
+            break;
+        }
+        case VX_TYPE_USER_DATA_OBJECT:
+        {
+            user_data_t user_data = {0};
+            ref = (vx_reference)vxCreateUserDataObject(context, "user_data_t", sizeof(user_data_t), &user_data);
+            break;
+        }
+        case VX_TYPE_REMAP:
+            ref = (vx_reference)vxCreateRemap(context, 16, 16, 16, 16);
+            break;
+        case VX_TYPE_THRESHOLD:
+            ref = (vx_reference)vxCreateThreshold(context, VX_THRESHOLD_TYPE_BINARY, VX_TYPE_UINT8); /* This will need updating!!)*/
+            break;
+        case TIVX_TYPE_RAW_IMAGE:
+            {
+                tivx_raw_image_create_params_t params = {
+                    .width = 16,
+                    .height = 16,
+                    .line_interleaved = vx_false_e,
+                    .meta_height_before = 1,
+                    .meta_height_after = 1,
+                    .num_exposures = 2,
+                    .format = {
+                        {.msb = 7, .pixel_container = TIVX_RAW_IMAGE_8_BIT},
+                        {.msb = 7, .pixel_container = TIVX_RAW_IMAGE_8_BIT},
+                        {.msb = 7, .pixel_container = TIVX_RAW_IMAGE_8_BIT}
+                    }
+                };
+                ref = (vx_reference)tivxCreateRawImage(context, &params);
+                break;
+            }
+        default:
+            printf("ERROR in TEST! Found unknown type %d (%s)\n", type, enumToString(type));
+    }
+    return ref;
+}
+
+TEST (supplementary_data, testRefCount)
+{
+    vx_context context = context_->vx_context_;
+
+    user_data_t user_data = {.numbers = {1, 2, 3, 4}};
+    vx_user_data_object exemplar = vxCreateUserDataObject(context, "user_data_t", sizeof(user_data_t), &user_data);
+    vx_status status = (vx_status)VX_SUCCESS;
+
+    /* First, set supplementary data on an object */
+    vx_reference test_object = (vx_reference)vxCreateImage(context, 10, 10, VX_DF_IMAGE_U8);
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxSetSupplementaryUserDataObject(test_object, exemplar));
+    VX_CALL(vxReleaseUserDataObject(&exemplar));
+
+    vx_reference test_object_copy = test_object;
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxRetainReference(test_object_copy));
+    vx_reference supp_data = (vx_reference)(vxGetSupplementaryUserDataObject(test_object, NULL, &status));
```

Comment:
Can we do an ASSERT_VX_OBJECT on all the references that we return from the vxGetSupplementaryUserDataObject calls?  It looks like there are several instances where this is missing

Reply:
we do check the status of the vxGetSupplementaryUserDataObject, is it not sufficient?
anyway I can add the check too.

Reply:
Our thought was that it could be theoretically possible that the status could be set properly while the object is still not proper, so it would provide a more robust test to assert this object is correct as well

Reply:
Ok I did it where the test is positive, for negative testing you can't assert the object.